packages/core/src/components/forms/numericInput.tsx

```diff
@@ -114,16 +114,18 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
     public constructor(props?: HTMLInputProps & INumericInputProps, context?: any) {
         super(props, context);
 
+        const { value } = props;
+
         this.state = {
             shouldSelectAfterUpdate: false,
-            value: this.getValueOrEmptyValue(props.value),
+            value: (value != null) ? value.toString() : NumericInput.VALUE_EMPTY,
```

Comment:
do you need to do the validity check here as well? since it's `number | string`?